### PR TITLE
Improving python installs

### PR DESCRIPTION
**Pull request by compass-dexter**
# Description

Not all packages support Python 3.12 yet. One notable dependency is aiohttp which currently has a [beta](https://github.com/aio-libs/aiohttp/issues/7675) release.

# Related issues

> If your PR responds to an issue for a bug fix or feature request, make sure to includes references to the issues (e.g. "fixes #xxxx").

# Point of attention

> Things you are not sure about that deserve special attention if you have doubts or questions.

### CODE REVIEW

File: sources/install/package_base.sh [hunk 1]
```diff
@@ -391,18 +388,18 @@ function package_base() {
...
-    pip3.6 install --upgrade pip
-    pip2 install --upgrade pip
-    pip3 install wheel
-    pip3.6 install wheel
-    pip2 install wheel
+    for v in $PYTHON_VERSIONS; do
+        pip${v} install --upgrade pip
+        pip${v} install wheel
+    done
     install_pipx
```
Comment (ShutdownRepo) [inline sources/install/package_base.sh:403]: I think pipx will be installed with the latest python3 here. Which will probably prevent `pipx --python python3.11` from working, what do you think? Have you tested?

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Thank you for this great PR! 
I think there will be other tools that don't support 3.11 yet and will need backporting.
Don't you think it's best to default to 3.11 for now, instead of having 3.12 and need to point to 3.11 for many tools?

**Comment by ShutdownRepo** [comment]
Opted for a static python3.11 version in https://github.com/ThePorgs/Exegol-images/commit/0bcaf25f525f0d93e17a743d235e9a9248922531 while we take care of this PR, so that the builds are not blocked in the mean time.

File: sources/install/package_base.sh (hunk 1 above)
Comment (compass-dexter) [reply sources/install/package_base.sh:403]: Yes, I tested that and it found `python3.11` in the pyenv installation.

**Comment by compass-dexter** [comment]
> Opted for a static python3.11 version in [0bcaf25](https://github.com/ThePorgs/Exegol-images/commit/0bcaf25f525f0d93e17a743d235e9a9248922531) while we take care of this PR, so that the builds are not blocked in the mean time.

That makes more sense for the time being. Some libraries will automatically add support for 3.12 at some point. Do you still want the other improvements of this PR (Typos and pyenv installation as loop)?

**Comment by ShutdownRepo** [comment]
> > Opted for a static python3.11 version in [0bcaf25](https://github.com/ThePorgs/Exegol-images/commit/0bcaf25f525f0d93e17a743d235e9a9248922531) while we take care of this PR, so that the builds are not blocked in the mean time.
> 
> That makes more sense for the time being. Some libraries will automatically add support for 3.12 at some point. Do you still want the other improvements of this PR (Typos and pyenv installation as loop)?

Absolutely, can you sync with latest changes, and keep the loop and typofixes? We'll add 3.12 in 2024 I think, when enough tool will support it

File: sources/install/package_base.sh (hunk 1 above)
Comment (ShutdownRepo) [reply sources/install/package_base.sh:409]: not applicable anymore since python3.12 is out of the picture for now

**Review by ShutdownRepo — APPROVED** [review]

**Pull request merged**